steering_controllers_library/CMakeLists.txt

```
@@ -0,0 +1,81 @@
+cmake_minimum_required(VERSION 3.8)
+project(steering_controllers_library)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+  control_msgs
+  controller_interface
+  geometry_msgs
+  hardware_interface
+  nav_msgs
+  pluginlib
+  rclcpp
+  rclcpp_lifecycle
+  realtime_tools
+  std_srvs
+  tf2
+  tf2_msgs
+  tf2_geometry_msgs
+  ackermann_msgs
+)
+
+find_package(ament_cmake REQUIRED)
+find_package(generate_parameter_library REQUIRED)
+
+foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
+  find_package(${Dependency} REQUIRED)
+endforeach()
+
+add_library(
+  ${PROJECT_NAME}
+  SHARED
+  src/steering_controllers_library.cpp
+  src/steering_odometry.cpp
+)
+
+generate_parameter_library(steering_controllers_library_parameters
+  config/steering_controllers_library.yaml
+)
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+target_link_libraries(${PROJECT_NAME}
+  steering_controllers_library_parameters)
+ament_target_dependencies(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_compile_definitions(${PROJECT_NAME} PRIVATE "STEERING_CONTROLLERS_BUILDING_DLL")
+
+install(
+  TARGETS
+  ${PROJECT_NAME}
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+)
+
+install(
+  DIRECTORY include/
+  #DESTINATION include/${PROJECT_NAME}   # TODO: it should be this, but then it doesn't compile (?)
+  DESTINATION include
+)
+
+if(BUILD_TESTING)
+  # TODO
```

@petkovich maybe adding a timeout test would make sense. @Robotgir can you help here, please?

```suggestion
  # Tested implicitly by controllers implementations.
```